components/web_discovery/browser/rsa.cc

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/web_discovery/browser/rsa.h"
+
+#include <vector>
+
+#include "base/base64.h"
+#include "third_party/boringssl/src/include/openssl/bn.h"
+#include "third_party/boringssl/src/include/openssl/bytestring.h"
+#include "third_party/boringssl/src/include/openssl/mem.h"
+#include "third_party/boringssl/src/include/openssl/rsa.h"
+
+namespace web_discovery {
+
+namespace {
+
+constexpr size_t kRsaKeySize = 2048;
+
+}  // namespace
+
+RSAKeyInfo::RSAKeyInfo() = default;
+RSAKeyInfo::~RSAKeyInfo() = default;
+
+std::unique_ptr<RSAKeyInfo> GenerateRSAKeyPair() {
```

Comment:
can we use `RsaHashedAlgorithm` from `webcrypto`?

Reply:
yes. i meant to swap this code out with `RsaHashedAlgorithm` before making the PR (i learned about its existence later on during implementation), but i think we can use it.

Reply:
or crypto::RSAPrivateKey?

Reply:
also what thread does this run on?

Reply:
all callers use a thread pool